src/components/ui/InfiniteList/index.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useRef, useEffect, useState, HTMLAttributes, ReactNode } from 'react';
+
+import { StyledInfiniteList } from './Styled';
+import { BaseProps } from '../Base';
+import { Cog as Spinner } from '../icons'; // TODO: update icon
+
+export interface InfiniteListProps extends Omit<HTMLAttributes<HTMLUListElement>, 'css'>, BaseProps {
+    /* The elements that will populat the list. These elements will be wrapped in <li> tags, so that can be any element. */
+    items: ReactNode[];
+    /* A callback function that will make an api call to load the next batch of items. */
+    handleLoad: () => void;
```

Comment:
If we're managing the loading state, this needs to be `async`

Reply:
Right, I'll update.